experimental/Schemes/DerivedPushforward.jl

```diff
@@ -46,6 +48,17 @@ function _derived_pushforward(M::SubquoModule)
   return st[1]
 end
 
+function _regularity_bound(c::AbsHyperComplex{T}, r::UnitRange) where {T<:ModuleFP}
+  isone(dim(c)) || error("complex must be one-dimensional")
+  isempty(r) && error("range must not be empty")
+  i0 = first(r)
+  result = maximum((x->degree(x; check=false)[1]).(gens(c[i0])))
+  for i in r
+    result = maximum(push!((x->degree(x; check=false)[1]).(gens(c[i])), result))
+  end
```

Comment:
General remark: If the free_resolution uses Singular and the input was graded/homogeneous, the regularity bound has already been computed and used internally.

On this code: I do not see the subtraction of i for the i-th module in the minimal free resolution here.  This takes the minimal shift in degrees into account to make the maps degree zero.  You have
regularity bound = smallest number s such that deg(e_{i,j}) \leq s + i, where i is the index of the module in the minimal free resolution and j refers to the j-th generator of this free module.
If you have been aiming at something else, then please do not call this regularity bound.

Reply:
This does not really belong to this PR. I will remove it.